VDA-XP-Reward/test/index.test.ts

```diff
@@ -0,0 +1,920 @@
+                })
+
+
+                it("Invalid year value", async () => {
+                    for (let n = 0; n < claim_gamer31_array.length; n++) {
+                        // Restore state
+                        await claimAvailableState.restore();
+        
+                        // Invalid Year
+                        const CLAIM_INVALID_YEAR: ClaimData = {...claim_gamer31_array[n]};
+                        const INVALID_YEAR_VALUES = [0, BLOCK_YEAR - 2, BLOCK_YEAR+1];
+                        for (let i = 0; i < INVALID_YEAR_VALUES.length; i++) {
+                            CLAIM_INVALID_YEAR.issueYear = INVALID_YEAR_VALUES[i];
+                            const claimData = [ generateClaimInfo(trustedSigners[0], registeredDID.address, CLAIM_INVALID_YEAR),];
+        
+                            await checkClaimXPReward(
+                                registeredDID,
+                                receiverAddress[0],
+                                claimData,
+                                false,
+                                "InvalidProofTime"
+                            )
+                        }
+                    }
+                })
+
+                it("Invalid month value", async () => {
+                    for (let n = 0; n < claim_gamer31_array.length; n++) {
+                        // Restore state
+                        await claimAvailableState.restore();
+
+                        const CLAIM_INVALID_MONTH: ClaimData = {...claim_gamer31_array[n]};
+                        // Test invalid month values - that are out of range
+                        const INVALID_MONTH_VALUES = [0, 13, 20];
+                        // Add month value that is 2 months before of current blockchain time
+                        if (BLOCK_MONTH > 2) {
+                            INVALID_MONTH_VALUES.push(BLOCK_MONTH - 2);
+                        }
+                        // Add month values that are after 1 and 5 months from current blockchain time
+                        INVALID_MONTH_VALUES.push((BLOCK_MONTH + 1) % 12, (BLOCK_MONTH + 5) % 12);
+
+                        CLAIM_INVALID_MONTH.issueYear = BLOCK_YEAR;
+                        for (let i = 0; i < INVALID_MONTH_VALUES.length; i++) {
+                            CLAIM_INVALID_MONTH.issueMonth = INVALID_MONTH_VALUES[i];
+                            const claimData = [ generateClaimInfo(trustedSigners[0], registeredDID.address, CLAIM_INVALID_MONTH),]
+        
+                            await checkClaimXPReward(
+                                registeredDID,
+                                receiverAddress[0],
+                                claimData,
+                                false,
+                                "InvalidProofTime"
+                            )
+                        }
+                    }
+        
+                })
+
+                it("Check in January", async () => {
```

Comment:
Please add a test to check that current blockchain is Jan, but claim was issued in December -- works correctly.

Reply:
Pls check the line#876:
![image](https://github.com/user-attachments/assets/9958a23b-9864-4b13-a029-9c1f330d0f0f)

Reply:
Resolved